pkg/model/participation/test/builder.go

```diff
@@ -0,0 +1,106 @@
+package test
+
+import (
+	"github.com/stretchr/testify/require"
+
+	"github.com/gohornet/hornet/pkg/model/hornet"
+	"github.com/gohornet/hornet/pkg/model/participation"
+	"github.com/iotaledger/hive.go/serializer"
+
+	"github.com/gohornet/hornet/pkg/model/utxo"
+	"github.com/gohornet/hornet/pkg/testsuite"
+	"github.com/gohornet/hornet/pkg/testsuite/utils"
+)
+
+type ParticipationHelper struct {
+	env                   *ParticipationTestEnv
+	wallet                *utils.HDWallet
+	msgBuilder            *testsuite.MessageBuilder
+	participationsBuilder *participation.ParticipationsBuilder
+}
+
+type SentParticipations struct {
+	builder *ParticipationHelper
+	message *testsuite.Message
+}
+
+func (env *ParticipationTestEnv) NewParticipationHelper(wallet *utils.HDWallet) *ParticipationHelper {
+	msgBuilder := env.te.NewMessageBuilder(participationIndexation).
+		LatestMilestonesAsParents()
+
+	return &ParticipationHelper{
+		env:                   env,
+		wallet:                wallet,
+		msgBuilder:            msgBuilder,
+		participationsBuilder: participation.NewParticipationsBuilder(),
+	}
+}
+
+func (b *ParticipationHelper) WholeWalletBalance() *ParticipationHelper {
+	b.msgBuilder.Amount(b.wallet.Balance())
+	return b
+}
+
+func (b *ParticipationHelper) Amount(amount uint64) *ParticipationHelper {
+	b.msgBuilder.Amount(amount)
+	return b
+}
+
+func (b *ParticipationHelper) Parents(parents hornet.MessageIDs) *ParticipationHelper {
+	require.NotEmpty(b.env.t, parents)
+	b.msgBuilder.Parents(parents)
+	return b
+}
+
+func (b *ParticipationHelper) UsingOutput(output *utxo.Output) *ParticipationHelper {
+	require.NotNil(b.env.t, output)
+	b.msgBuilder.UsingOutput(output)
+	return b
+}
+
+func (b *ParticipationHelper) AddParticipations(participations []*participation.Participation) *ParticipationHelper {
+	require.NotEmpty(b.env.t, participations)
+	for _, p := range participations {
+		b.AddParticipation(p)
+	}
+	return b
+}
+
+func (b *ParticipationHelper) AddDefaultBallotVote(eventID participation.EventID) *ParticipationHelper {
```

Comment:
Is this function name still correct, or is this a `AddDefaultParticipation` ?

Reply:
Yes, it explicitely adds a vote for a ballot